dot4gravity/src/lib.rs

```diff
@@ -1,8 +1,707 @@
+        self.row < BOARD_WIDTH && self.col < BOARD_HEIGHT
+    }
+
+    /// Tells if a cell is in the oposite of a side.
+    fn is_opposite_cell(&self, side: Side) -> bool {
+        match side {
+            Side::North => self.row == BOARD_HEIGHT - 1,
+            Side::East => self.col == 0,
+            Side::South => self.row == 0,
+            Side::West => self.col == BOARD_WIDTH - 1,
+        }
+    }
+}
+
+/// Sides of the board from which a player can drop a stone.
+#[derive(Clone, Debug)]
+enum Side {
+    North,
+    East,
+    South,
+    West,
+}
+
+#[derive(Clone, Debug)]
+struct Board {
+    cells: [[Cell; BOARD_WIDTH]; BOARD_HEIGHT],
+}
+
+impl Board {
+    pub fn new() -> Board {
+        Board {
+            cells: [[Cell::Empty; BOARD_WIDTH]; BOARD_HEIGHT],
+        }
+    }
+
+    fn get_cell(&self, position: &Coordinates) -> Cell {
+        self.cells[position.row][position.col]
+    }
+
+    fn change_cell(&mut self, position: Coordinates, cell: Cell) {
+        self.cells[position.row][position.col] = cell;
+        assert_eq!(self.cells[position.row][position.col], cell);
+    }
+
+    fn populate_with_random_blocks(mut board: Board, mut num_of_blocks: usize) -> Board {
+        let mut blocks = Vec::<Coordinates>::new();
+        let mut rng = rand::thread_rng();
+
+        while num_of_blocks > 0 {
+            let block_coordinates = Coordinates {
+                row: rng.gen_range(0..BOARD_HEIGHT),
+                col: rng.gen_range(0..BOARD_WIDTH),
+            };
+            if !blocks.contains(&block_coordinates) {
+                num_of_blocks -= 1;
+                blocks.push(block_coordinates)
+            }
+        }
+        for block_coordinates in blocks {
+            board.change_cell(block_coordinates.clone(), Cell::Block);
+        }
+        board
+    }
+
+    fn explode_bomb(mut board: Board, bomb_position: Coordinates) -> Board {
+        let offsets: [(i8, i8); 9] = [
+            (0, 0),
+            (-1, -1),
+            (0, -1),
+            (1, -1),
+            (1, 0),
+            (1, 1),
+            (0, 1),
+            (-1, 1),
+            (-1, 0),
+        ];
+
+        // Collect the explodable cells around.
+        let explodable_cells_coords: Vec<Coordinates> = offsets
+            .iter()
+            .map(|offset| {
+                (
+                    offset.0 + bomb_position.row as i8,
+                    offset.1 + bomb_position.col as i8,
+                )
+            })
+            .filter(|offset| offset.0 >= 0 && offset.1 >= 0)
+            .map(|offset| Coordinates {
+                row: offset.0 as usize,
+                col: offset.1 as usize,
+            })
+            .filter(|position| {
+                position.is_inside_board() && board.get_cell(&position).is_explodable()
+            })
+            .collect();
+
+        for position in explodable_cells_coords {
+            board.change_cell(position, Cell::Empty);
+        }
```

Comment:
It might be more concise to do just one loop:
```suggestion
        offsets
            .iter()
            .map(|(row_offset, col_offset)| Coordinates {
                row: (row_offset + bomb_position.row as i8) as usize,
                col: (col_offset + bomb_position.col as i8) as usize,
            })
            .for_each(|position| {
                if position.is_inside_board()
                    && board.get_cell(&position).is_explodable()
                    && position.row >= 0
                    && position.col >= 0
                {
                    board.change_cell(position, Cell::Empty);
                }
            });
```